packages/app/src/cli/services/deploy/mode.ts

```diff
@@ -0,0 +1,129 @@
+import {OrganizationApp} from '../../models/organization.js'
+import {DeployContextOptions} from '../context.js'
+import {SetBetaFlagQuery, SetBetaFlagSchema, SetBetaFlagVariables} from '../../api/graphql/set_beta_flag.js'
+import {PackageManager} from '@shopify/cli-kit/node/node-package-manager'
+import {formatPackageManagerCommand, outputCompleted, outputNewline} from '@shopify/cli-kit/node/output'
+import {InfoMessage, renderConfirmationPrompt, renderInfo, renderTasks, renderWarning} from '@shopify/cli-kit/node/ui'
+import figures from '@shopify/cli-kit/node/figures'
+import {partnersRequest} from '@shopify/cli-kit/node/api/partners'
+import {BugError} from '@shopify/cli-kit/node/error'
+
+export type DeploymentMode = 'legacy' | 'unified' | 'unified-skip-release'
+
+export async function getDeploymentMode(app: OrganizationApp, options: DeployContextOptions, token: string) {
+  let deploymentMode: DeploymentMode = app.betas?.unifiedAppDeployment ? 'unified' : 'legacy'
+  deploymentMode = await upgradeDeploymentMode(deploymentMode, options.app.packageManager, app, token)
```

Comment:
is this a getter but also automatically upgrades you `unified`?

Reply:
yes, it's a bit confusing. Let me refactor this to be clearer

Reply:
@isaacroldan I talked to @ani-shopify and they would like to use this in tomorrow's Build Convergence demos. I have done a little refactor and It looks clearer for me. 
Anyway, we could create a different PR to improve the code quality if we decide it's really needed